### PR TITLE
[Hub Generated] Review request for Microsoft.Insights to add version stable/2015-05-01

### PR DESCRIPTION
**Pull request by arolshan**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/arolshan/azure-rest-api-specs/dev-applicationinsights-Microsoft.Insights-2015-05-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by ArcturusZhang** [comment]
Looks good to me, please wait for ARM team's response

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json [hunk 1]
```diff
@@ -486,6 +486,11 @@
           "externalDocs": {
             "url": "https://aka.ms/sl0yqd"
           }
+        },
+        "RetentionInDays": {
+          "type": "integer",
+          "description": "Retention period by number of days.",
```
Comment (yoramsinger) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json]: Retention period in days.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json [hunk 2]
```diff
@@ -486,6 +486,11 @@
           "externalDocs": {
             "url": "https://aka.ms/sl0yqd"
           }
+        },
+        "RetentionInDays": {
```
Comment (KrisBash) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json]: Adding a writeable property within an API version is considered a breaking change. Why is this not going in a new version?

**Review by KrisBash — CHANGES_REQUESTED** [review]
see comment

**Comment by arolshan** [comment]
Hi @Kristopher Bash<mailto:krbash@microsoft.com>,
Can you please explain how is this breaking change?
The property is optional and ignored if empty.
Best,
Arik.
________________________________

**Comment by arolshan** [comment]
From our breaking change<https://microsoft.sharepoint.com/:w:/t/azureresourcemanagerteam/EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw?e=k7RZpV> guide:
New property added to response

If a new property/field is added to the response an API, the GET-PUT pipeline will be broken. Consider the case where from portal a customer updates the value of a new property "A". Another customer does a GET of this resource using the SDK. The SDK will ignore the property since it does not understand it. From the SDK, the customer does a PUT using the model that was returned from the GET. This will overwrite the change made by the first customer from the portal.

By adding within a version, a mix of current and downlevel clients will exist in the world, and it is ambiguous whether an omission of the property is due to intentional removal or legacy sdk.

-Kris

**Comment by arolshan** [comment]
Okay, thanks for the clarification, I'll update the version. Is there anything else I need to do ?

From: Kristopher Bash <krbash@microsoft.com>
Sent: Wednesday, October 16, 2019 8:42 PM
To: Arik Olsh <Arie.Olshanezky@microsoft.com>; Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>; Azure/azure-rest-api-specs <reply@reply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [Azure/azure-rest-api-specs] [Hub Generated] Review request for Microsoft.Insights to add version stable/2015-05-01 (#7455)

From our breaking change<https://nam06.safelinks.protection.outlook.com/ap/w-59584e83/?url=https%3A%2F%2Fmicrosoft.sharepoint.com%2F%3Aw%3A%2Ft%2Fazureresourcemanagerteam%2FEWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw%3Fe%3Dk7RZpV&data=02%7C01%7CArie.Olshanezky%40microsoft.com%7Cfb9dfc9b164b4d19710c08d752602f05%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637068445374210131&sdata=GzQVT9%2B7%2B2ySBDXT5vhbOWjvHvvdXPpdK0CM7XnKQww%3D&reserved=0> guide:
New property added to response

If a new property/field is added to the response an API, the GET-PUT pipeline will be broken. Consider the case where from portal a customer updates the value of a new property "A". Another customer does a GET of this resource using the SDK. The SDK will ignore the property since it does not understand it. From the SDK, the customer does a PUT using the model that was returned from the GET. This will overwrite the change made by the first customer from the portal.

By adding within a version, a mix of current and downlevel clients will exist in the world, and it is ambiguous whether an omission of the property is due to intentional removal or legacy sdk.

-Kris

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json (hunk 2 above)
Comment (arolshan) [reply specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/components_API.json]: Hi Kris
please refer to my update PR. Created new one with suitable name.

https://github.com/Azure/azure-rest-api-specs/pull/7539

**Pull request closed without merging**